core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -316,6 +332,46 @@ public ServerBuilder port(ServerPort port) {
         return this;
     }
 
+    /**
+     * Sets the {@link ChannelOption} of the server socket bound by {@link Server}.
+     * Note that the previously added option will be overridden if the same option is set again.
+     *
+     * <pre>{@code
+     * ServerBuilder sb = new ServerBuilder();
+     * sb.channelOption(ChannelOption.BACKLOG, 1024)
+     *   .channelOption(ChannelOption.WRITE_BUFFER_WATER_MARK, 32 * 1024);
+     * }</pre>
+     */
+    public <T> ServerBuilder channelOption(ChannelOption<T> option, T value) {
+        requireNonNull(option, "option");
+        checkArgument(!PROHIBITED_SOCKET_OPTIONS.contains(option),
+                      "prohibited socket option: %s", option);
+
+        option.validate(value);
+        this.channelOptions.put(option, value);
+        return this;
+    }
+
+    /**
+     * Sets the {@link ChannelOption} of server sockets accepted by {@link Server}.
```

Comment:
server sockets -> the sockets (because strictly speaking the accepted sockets are not server sockets)